crates/biome_module_graph/src/js_semantic_model.rs

```diff
@@ -0,0 +1,309 @@
+    pub declarations: Box<[JsDeclaration]>,
+}
+
+impl JsSemanticModel {
+    /// Finds the value declaration with the given `name`.
+    pub fn get_value(&self, name: &str) -> Option<&JsDeclaration> {
+        self.declarations
+            .iter()
+            .find(|decl| decl.name == name && decl.kind.might_be_value())
+    }
+
+    /// Finds the type declaration with the given `name`.
+    pub fn get_type(&self, name: &str) -> Option<&JsDeclaration> {
+        self.declarations
+            .iter()
+            .find(|decl| decl.name == name && decl.kind.might_be_type())
+    }
+}
+
+/// A single declaration of a variable or type.
+#[derive(Clone, Debug)]
+pub struct JsDeclaration {
+    /// The name of the thing being declared.
+    pub name: Text,
+
+    /// The kind of thing being declared.
+    pub kind: JsDeclarationKind,
+
+    /// Optional JSDoc comment associated with the declaration.
+    pub jsdoc_comment: Option<JsdocComment>,
+}
+
+#[derive(Clone, Debug)]
+pub enum JsDeclarationKind {
+    /// A `class` declaration.
+    ///
+    /// The type defined by the class.
+    #[expect(unused)] // TODO
+    Class(Type),
+
+    /// A `function` or `var` declaration.
+    ///
+    /// The type (either declared or inferred) of the value.
+    HoistedValue(Type),
+
+    /// An `import` declaration.
+    Import(Text),
+
+    /// An `import type` declaration.
+    ImportType(Text),
+
+    /// An interface declaration.
+    #[expect(unused)] // TODO
+    Interface,
+
+    /// A module declaration.
+    #[expect(unused)] // TODO
+    Module,
+
+    /// A namespace declaration.
+    #[expect(unused)] // TODO
+    Namespace,
+
+    /// A type declaration.
+    Type(Type),
+
+    /// A `let` or `const` declaration.
+    ///
+    /// The type (either declared or inferred) of the value.
+    Value(Type),
+}
+
+impl JsDeclarationKind {
+    /// Returns `true` for any declaration that _may_ be a type.
+    fn might_be_type(&self) -> bool {
+        matches!(
+            self,
+            Self::Class(_)
+                | Self::Import(_)
+                | Self::ImportType(_)
+                | Self::Namespace
+                | Self::Interface
+                | Self::Type(_)
+        )
+    }
+
+    /// Returns `true` for any declaration that _may_ be a value.
+    fn might_be_value(&self) -> bool {
+        matches!(
+            self,
+            Self::Class(_) | Self::HoistedValue(_) | Self::Import(_) | Self::Value(_)
+        )
+    }
+}
+
+/// Responsible for building the JsSemanticBuilder.
+#[derive(Debug, Default)]
+pub struct JsSemanticModelBuilder {
+    declarations: Vec<JsDeclaration>,
+}
+
+impl JsSemanticModelBuilder {
+    pub fn build(self) -> JsSemanticModel {
+        JsSemanticModel {
+            declarations: self.declarations.into(),
+        }
+    }
+
+    pub fn push_node(&mut self, node: &JsSyntaxNode) {
+        if let Some(import) = AnyJsImportClause::cast_ref(node) {
+            self.push_import(import);
```

Comment:
You should use `AnyJsImportLike`, so you can handle `require()` and `import()` too

Reply:
I'll add a TODO for it.